### DISCUSSION TITLE
Episode 195: 🇨🇦 CppNorth Live 🇨🇦 David Olsen & Pure Chaos!

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#94] [Episode]
Episode Link: https://adspthepodcast.com/2024/08/16/Episode-195.html

Feel free to write comments or ask questions 🙂

**Comment by joblobob** [comment]
Best lines of the podcast! 
-Bryce: _"Mike Daum, would you like to say anything to the ADSP audience?"_
-Mike: _"You belong at CppNorth."_ 🤟 


-Ben Deane: _"Not a C++ Cabin?"_
-Bryce: _"You know what? Ben deserves to takeover from me, just for that one!"_ 😄  
-Ben Deane: _"If elected, I pledge to bring actual algorithms and data structures to this podcast."_ 

Thank you Anti-Bryce!! I vote for you ✌️ 

In other news.... I hope to see you next year Conor, I haven't had time to have a chat with you during the conference!

**Comment by koenpoppe** [comment]
Allow me to elaborate on my question because it's not exactly about the algorithm being two phase or not. I was wondering if an iterator that only visits the unique elements of the original array (and jumps over the duplicate ones) could be less expensive than one that moves all unique elements to the front of the array. It's an idea that came to me whilst attempting to remove the temporary array that is needed in the CUB implementation, indeed, as Bryce and episode 134 mentioned. 

I’ve written it up with a small proof of concept [UniqueView](https://github.com/koenpoppe/UniqueView) that is more performant for at least some input data.

PS. many thanks to Tristan for bringing this up and allowing me to contribute to the chaos =)

**Comment by mrpenner** [comment]
I vote for more algorithms and data structures and less chaos, whether from Bryce or Ben Deane. I have plenty of other sources for chaos. :) If you're going for entertainment value though, include both Bryce and Ben Deane. They bounce well off each other!

**Comment by marcosbento** [comment]
Love algorithms and data structure stuff... but it wouldn't be the same without the pure chaos. Keep going strong!

Loved the episode, and... Congratulations, Conor!
